sequence_task_3/student/include/impl.h

```diff
@@ -8,45 +8,100 @@ class DynamicPlaylist
 {
 public:
     /** @todo Member traits */
-
+    using value_type = typename Container::value_type;
+    using reference = typename Container::reference;
+    using const_reference = typename Container::const_reference;
+    using iterator = typename Container::iterator;
+    using const_iterator = typename Container::const_iterator;
+    using difference_type = typename Container::difference_type;
+    using size_type = typename Container::size_type;
     /** @todo Iterators */
+    const_iterator begin() const {
+        return m_tracklist.cbegin();
+    }
+
+    const_iterator end() const {
+        return m_tracklist.cend();
+    }
 
     DynamicPlaylist() = default;
 
     /** @todo Constructor from two iterators*/
-
+    template<typename iterator>
+    DynamicPlaylist(iterator first, iterator last)
```

Comment:
use constructor directly and initialize instead of assignment

Reply:
Can you explain to me in more detail, please?

Reply:
https://www.learncpp.com/cpp-tutorial/constructor-member-initializer-lists/

Here, you are doing two operations:
m_tracklist {}
m_tracklist.assign(first,last)

Why can't we merge whem?

Reply:
I understand the initialization, but how am I able to use the iterators afterwards?

Reply:
You can use constructor that accepts two iterators: m_tracklist {first, last};